com.unity.render-pipelines.universal/Editor/Converter/Converters.cs

```diff
@@ -0,0 +1,226 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace UnityEditor.Rendering.Universal
+{
+    /// <summary>
+    /// Used for Converting in Batch Mode.
+    /// </summary>
+    public enum ConverterFilter
+    {
+        Inclusive,
+        Exclusive
+    }
+
+    /// <summary>
+    /// Used for Converting in Batch Mode.
+    /// </summary>
+    public enum ConverterContainerId
+    {
+        BuiltInToURP,
+        BuiltInToURP2D,
+        UpgradeURP2DAssets,
+    }
+
+    /// <summary>
+    /// Used for Converting in Batch Mode.
+    /// </summary>
+    public enum ConverterId
+    {
+        Material,
+        RenderSettings,
+        AnimationClip,
+        ReadonlyMaterial,
+        PPv2,
+        ParametricToFreeformLight,
+    }
+
+    public static class Converters
+    {
+        static Type GetContainerType(ConverterContainerId containerName)
+        {
+            switch (containerName)
+            {
+                case ConverterContainerId.BuiltInToURP:
+                    return typeof(BuiltInToURPConverterContainer);
+                case ConverterContainerId.BuiltInToURP2D:
+                    return typeof(BuiltInToURP2DConverterContainer);
+                case ConverterContainerId.UpgradeURP2DAssets:
+                    return typeof(UpgradeURP2DAssetsContainer);
+            }
+
+            return null;
+        }
+
+        static Type GetConverterType(ConverterId converterName)
+        {
+            switch (converterName)
+            {
+                case ConverterId.Material:
+                    return typeof(UniversalRenderPipelineMaterialUpgrader);
+                case ConverterId.RenderSettings:
+                    return typeof(RenderSettingsConverter);
+                case ConverterId.AnimationClip:
+                    return typeof(AnimationClipConverter);
+                case ConverterId.ReadonlyMaterial:
+                    return typeof(ReadonlyMaterialConverter);
+#if PPV2_EXISTS
+                case ConverterId.PPv2:
+                    return typeof(PPv2Converter);
+#endif
+                case ConverterId.ParametricToFreeformLight:
+                    return typeof(ParametricToFreeformLightUpgrader);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// The method that will be run when converting assets in batch mode.
```

Comment:
This reads kinds of weird. Isn't this a method that the user themselves would be calling?